portal/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -0,0 +1,54 @@
+package cromwell.pipeline.service
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.server.Route
+import cromwell.pipeline.datastorage.dao.repository.UserRepository
+import cromwell.pipeline.datastorage.dto.user.UpdateRequest
+import cromwell.pipeline.datastorage.dto.user.{ UpdatePasswordRequest, UpdateRequest }
+import cromwell.pipeline.datastorage.dto.{ User, UserId }
+import cromwell.pipeline.utils.StringUtils
+import play.api.libs.json.JsError
+import play.api.libs.json.JsResult.Exception
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Random
+
+class UserService(userRepository: UserRepository)(implicit executionContext: ExecutionContext) {
+
+  def updateUser(id: String, request: UpdateRequest): Future[Int] = {
+    val updatedUser = User(
+      userId = UserId(id),
+      email = request.email,
+      passwordSalt = "",
+      passwordHash = "",
+      firstName = request.firstName,
+      lastName = request.lastName
+    )
+    userRepository.updateUser(updatedUser)
+  }
+
+  def updateUserPassword(id: String, request: UpdatePasswordRequest): Future[Int] =
+    if (request.newPassword != request.repeatPassword) {
+      throw Exception(JsError("different password"))
```

Comment:
It's good you're handling this case. 
However, in Scala we don't like throwing exceptions so much. 
Consider using `Future.failed`. 
Throwing `Exception` is also very abstract. You need to use something more specific. If you want, you can create your own exception

Reply:
done